pyomo/repn/plugins/nl_writer.py

```diff
@@ -85,6 +85,15 @@
 )
 
 
+class NLWriterEmptyModelError(ValueError):
+    """
+    A custom exception to allow handling of a
+    model with no free variables.
+    """
+
+    pass
```

Comment:
Suggestion here: We have a "custom" exception from `pyomo/common/errors` called `PyomoException` that you might want to inherit from instead. We do this fairly frequently for custom error types so it is also clear when an exception comes from Pyomo as opposed to something else (e.g., `gurobipy` or `idaes`).

Example:
```
from pyomo.common.errors import PyomoException

class MyCustomError(PyomoException):
   """
   My custom error
   """
```

Also note that you don't need to put `pass` as long as you have a docstring.

Reply:
Fixed, I initially had it inheriting from `ValueError` since that was raised previously, but I don't suppose there is much need for backwards compatibility in this case!

Reply:
Nevermind, there were a couple of tests designed to catch this - I have replaced the asserted exception in those tests